sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/ingestion/models/CustomPropertiesLog.java

```diff
@@ -0,0 +1,193 @@
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TimeZone;
+
+/**
+ * The custom properties log model.
+ */
+public class CustomPropertiesLog extends AbstractLog {
+
+    /**
+     * Log type.
+     */
+    public static final String TYPE = "custom_properties";
+
+    private static final String PROPERTIES = "properties";
+
+    private static final String PROPERTY_TYPE = "type";
+
+    private static final String PROPERTY_NAME = "name";
+
+    private static final String PROPERTY_VALUE = "value";
+
+    private static final String PROPERTY_TYPE_CLEAR = "clear";
+
+    private static final String PROPERTY_TYPE_BOOLEAN = "boolean";
+
+    private static final String PROPERTY_TYPE_NUMBER = "number";
+
+    private static final String PROPERTY_TYPE_DATETIME = "date_time";
+
+    private static final String PROPERTY_TYPE_STRING = "string";
+
+    private static final ThreadLocal<DateFormat> DATETIME_FORMAT = new ThreadLocal<DateFormat>() {
+
+        @Override
+        protected DateFormat initialValue() {
+            DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US);
+            dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+            return dateFormat;
+        }
+    };
+
+    /**
+     * Properties key/value pairs.
+     */
+    private Map<String, Object> properties;
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    /**
+     * Get the properties value.
+     *
+     * @return the properties value
+     */
+    public Map<String, Object> getProperties() {
+        return this.properties;
+    }
+
+    /**
+     * Set the properties value.
+     *
+     * @param properties the properties value to set
+     */
+    public void setProperties(Map<String, Object> properties) {
+        this.properties = properties;
+    }
+
+    @Override
+    public void read(JSONObject object) throws JSONException {
+        super.read(object);
+        setProperties(readProperties(object));
+    }
+
+    @Override
+    public void write(JSONStringer writer) throws JSONException {
+        super.write(writer);
+        writeProperties(writer, getProperties());
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
+
+        CustomPropertiesLog that = (CustomPropertiesLog) o;
+
+        return properties != null ? properties.equals(that.properties) : that.properties == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = super.hashCode();
+        result = 31 * result + (properties != null ? properties.hashCode() : 0);
+        return result;
+    }
+
+    private static Map<String, Object> readProperties(JSONObject object) throws JSONException {
+        JSONArray jArray = object.optJSONArray(PROPERTIES);
```

Comment:
I meant use `getJSONArray` instead of `opt`, `get` triggers a the JSONException if null for you.